backend/core/src/main.rs

```diff
@@ -103,6 +105,27 @@ async fn node_route(
     )))
 }
 
+#[get("/shard_submit/{chain_hash}")]
+async fn shard_route(
+    req: HttpRequest,
+    stream: web::Payload,
+    aggregator: web::Data<Addr<Aggregator>>,
+    path: web::Path<Box<str>>,
+) -> Result<HttpResponse, Error> {
+    let hash_str = path.into_inner();
+    let genesis_hash = hash_str.parse()?;
+
+    let mut res = ws::handshake(&req)?;
+
+    let aggregator = aggregator.get_ref().clone();
+
+    Ok(res.streaming(ws::WebsocketContext::with_codec(
+        ShardConnector::new(aggregator, genesis_hash),
+        stream,
+        Codec::new().max_size(10 * 1024 * 1024), // 10mb frame limit
```

Comment:
Use the `CONT_BUF_LIMIT` constant here?

Reply:
DQ: What is the significance of the 10mb frame limit?

Reply:
Websocket communication is frame-based, i.e. the payload is split into chunks of convenient size. Here we're saying that to us "convenient" means "less than 10mb".
The [gory details](https://tools.ietf.org/html/rfc6455#section-5.2).

Reply:
Ahh, okay thanks for the description make sense, and those docs are gold thank you very much!